sympy/solvers/solveset.py

```diff
@@ -548,6 +528,9 @@ def _solveset(f, symbol, domain, _check=False):
         a, h = f.as_independent(symbol)
         m, h = h.as_independent(symbol, as_Add=False)
         f = a/m + h  # XXX condition `m != 0` should be added to soln
+    numer, denom = f.as_numer_denom()
```

Comment:
My guess is that the correct behavior will be more complicated than this, because there could be denominators hidden in an expression (I think @smichr has some good examples).